test/corpus/prec.txt

```diff
@@ -0,0 +1,1528 @@
+
+(haskell
+  (imports
+    (import
+      (module)
+      (import_list
+        (import_item
+          (operator))))))
+
+================================================================================
+prec: varop export with leading dot
+================================================================================
+
+module A ((.=)) where
+
+--------------------------------------------------------------------------------
+
+(haskell
+  (header
+    (module)
+    (exports
+      (export
+        (operator)))))
+
+================================================================================
+prec: infix varop with leading dot in list
+================================================================================
+
+a = [a .+ a]
+a = [a ..+ a]
+a = [a A..+ a]
+
+--------------------------------------------------------------------------------
+
+(haskell
+  (declarations
+    (function
+      (variable)
+      (exp_list
+        (exp_infix
+          (exp_name
+            (variable))
+          (operator)
+          (exp_name
+            (variable)))))
+    (function
+      (variable)
+      (exp_list
+        (exp_infix
+          (exp_name
+            (variable))
+          (operator)
+          (exp_name
+            (variable)))))
+    (function
+      (variable)
+      (exp_list
+        (exp_infix
+          (exp_name
+            (variable))
+          (qualified_operator
+            (module)
+            (operator))
+          (exp_name
+            (variable)))))))
+
+================================================================================
+prec: tight infix varop with leading dot
+================================================================================
+
+a = a.+a
+a = a..+a
+
+--------------------------------------------------------------------------------
+
+(haskell
+  (declarations
+    (function
+      (variable)
+      (exp_infix
+        (exp_name
+          (variable))
+        (operator)
+        (exp_name
+          (variable))))
+    (function
+      (variable)
+      (exp_infix
+        (exp_name
+          (variable))
+        (operator)
+        (exp_name
+          (variable))))))
+
+================================================================================
+prec: arithmetic sequence
+================================================================================
+
+a = [1..]
+a = [a..]
+a = [1 ..]
+a = [a ..]
+
+a = [1..2]
+a = [a..a]
+a = [1 ..2]
+a = [a ..a]
+a = [1.. 2]
+a = [a.. a]
+a = [1 .. 2]
+a = [a .. a]
+
+a = [1,2..]
+a = [a,a..]
+a = [1,2 ..]
+a = [a,a ..]
+
+a = [1,2..3]
+a = [a,a..a]
+a = [1,2 ..3]
+a = [a,a ..a]
+a = [1,2.. 3]
+a = [a,a.. a]
+a = [1,2 .. 3]
+a = [a,a .. a]
+
+-- This should be an error, since a conid followed by two dots parses as the
+-- qualified composition operator, but we allow it since there's no way to use
+-- an operator here, like it would be in a section.
+a = [A..a]
```

Comment:
Very nice.

Next step is to fix this in GHC's parser, yeah?

```
λ> [False..True]

<interactive>:1:2: error:
    Not in scope: ‘False..’
```

Reply:
ah 😅 GHC makes up for it by permitting `..` to be referred to as an operator if you use a qualifying module, like `a A... a` 😁